internal/services/storage/client/data_plane.go

```diff
@@ -46,7 +53,11 @@ func (Client) DataPlaneOperationSupportingOnlySharedKeyAuth() DataPlaneOperation
 
 func (c Client) configureDataPlane(ctx context.Context, clientName, resourceIdentifier string, baseClient client.BaseClient, account accountDetails, operation DataPlaneOperation) error {
 	if operation.SupportsAadAuthentication && c.authConfigForAzureAD != nil {
-		api := c.authConfigForAzureAD.Environment.Storage.WithResourceIdentifier(resourceIdentifier)
+		api := c.authConfigForAzureAD.Environment.Storage
```

Comment:
The resource identifier is required for all data plane APIs, as it specifies the hostname for the API - which in all cases is the relevant URL of the particular service under the storage account subdomain. Why is it left unconfigured here? This will simply cause the request(s) to fail?

Reply:
So I looked into the complete coding again and I think I can drop most of the parts.
But the important part that fixes the "Network" and "AccessKey" is that we use an "unconfigured" api here.

When we use the "generic" api here, we get the "Azure AD" scope that is required to access the storage account without the "access keys" or a "secured" network.

The required scope that is required to access the storage account is:
`"https://storage.azure.com/.default"` it's also the default at the AzureSDK and the AzureCLI.

When we use the `WithResourceIdentifiere` we will set the scope like this:
`https://<MY_CONTAINER>.blob.core.windows.net/.default`

IMHO I think we can switch the auth SCOPE to the `generic once`

Reply:
Interesting, can you explain which scenarios require the generic scope? How does this relate to whether the Terraform host is included/excluded in any network rules? [The docs](https://learn.microsoft.com/en-us/azure/storage/blobs/authorize-access-azure-active-directory#microsoft-authentication-library-msal) imply these are interchangeable. Whilst the SDK does default to the global scope for Storage, to date we've intentionally set the narrowest scope within the provider, in accordance with the principle of least privilege.

Reply:
@manicminer 
I also couldn't find anything at the docs why we need to use the `generic` scope... but will ask someone to get more information about that.

What I can say is that the ` az cli` with `--auth-mode login` and the `azcopy` tool always use the `generic`scope.
https://github.com/Azure/azure-storage-azcopy/blob/0475829f98233f7176a62dbac50d36e3d90e83d0/common/oauthTokenManager.go#L55